src/librustc/ty/mod.rs

```diff
@@ -2057,8 +2058,8 @@ impl_stable_hash_for!(struct ReprFlags {
 #[derive(Copy, Clone, Debug, Eq, PartialEq, RustcEncodable, RustcDecodable, Default)]
 pub struct ReprOptions {
     pub int: Option<attr::IntType>,
-    pub align: u32,
-    pub pack: u32,
+    pub align: Option<Align>,
```

Comment:
This can just be `Align`, although this is probably okay too (cc @oli-obk).

Reply:
That's what I tried initially, but some code in ` src/librustc_typeck/check/mod.rs` explicitly checks if an alignment is set.

Reply:
We could also use an enum

```rust
enum AlignRepr {
    Align(Align),
    Packed(Align),
}
```

To encode the fact that there can be only one of them.